src/Filter/StringFilter.php

```diff
@@ -165,17 +146,19 @@ public function filter(BaseProxyQueryInterface $query, $alias, $field, $data)
             $parameterName,
             sprintf(
                 $format,
-                false !== $caseSensitive || '' === $data['value'] ? $data['value'] : mb_strtolower($data['value'])
+                true !== $forceCaseInsensitivity || '' === $data['value'] ? $data['value'] : mb_strtolower($data['value'])
             )
         );
     }
 
     public function getDefaultOptions()
     {
         return [
-            // NEXT_MAJOR: Remove the format option.
+            // NEXT_MAJOR: Remove the "format" option.
             'format' => '%%%s%%',
+            // NEXT_MAJOR: Remove the "case_sensitive" option.
             'case_sensitive' => null,
+            'force_case_insensitivity' => null,
```

Comment:
Why null ?
What's the difference between null and false ?

If I understand the option, the only allowed value would be `false` and `true`

Reply:
We haven't other way to detect if this option is set or not. Currently, I'm using the value from "case_sensitive" if this option is not set.
I think we have 2 choices: add a `hasOption()` method, or leave this value adding a `NEXT_MAJOR` comment.
WDYT?

Reply:
Why do you need to know if the option is set ?

If `case_sensitive` is not null, use this (deprecated) value
If `case_sensitive` is null, use force_case_insensitivity, which is by default to false.

Reply:
Because I'd like the new option to be respected if it is set, regardless the value of the deprecated option.

Reply:
Then we could add a NEXT_MAJOR comment to change the default value to `false` and remove the `true ===` or `true !==` checks. But It feel weird to provide a `null` value and a `false` value which has the same behavior. And introducing a `null` default value which is already deprecated and will be changed to false in the next major...

But when I read `force_case_insensitivity => false, case_sensitive => false`:
I would expect
- `force_case_insensitivity` won't force the case insensitivity so it will do nothing. (It never was said that `false` will force the case sensitivity)
- `case_sensitive` will force the case insensitivity because it's already the behavior of this option

Reply:
>But when I read force_case_insensitivity => false, case_sensitive => false:
I would expect
>
>force_case_insensitivity won't force the case insensitivity so it will do nothing. (It never was said that false will force the case sensitivity)
case_sensitive will force the case insensitivity because it's already the behavior of this option

I think the idea transmitted by the name `case_sensitive` leads to wrong assumptions. In fact, that's the reason for this change.
That's why I feel like a more natural behavior to completely ignore the option if its counterpart (`force_case_insensitivity`) is set.

Reply:
The `NEXT_MAJOR` comment about the default value was added.

Reply:
@VincentLanglet, I've added a comment in the upgrade document to make this behavior explicit.